tcp/clientobserve.go

```diff
@@ -79,7 +79,9 @@ func (o *Observation) handler(w *ResponseWriter, r *pool.Message) {
 }
 
 func (o *Observation) cleanUp() bool {
-	o.cc.observationTokenHandler.Pop(o.token)
+	if _, err := o.cc.observationTokenHandler.Pop(o.token); err != nil {
```

Comment:
ignore error - it means that observation does not exist at all.

Reply:
`o.cc.observationRequests.PullOut` should still be called in that case?

Reply:
Updated